packages/core/components/Modal/MoveFileManifest/index.tsx

```diff
@@ -0,0 +1,97 @@
+import * as React from "react";
+import { useSelector } from "react-redux";
+
+import { ModalProps } from "..";
+import BaseModal from "../BaseModal";
+import { PrimaryButton } from "../../Buttons";
+import { interaction, selection } from "../../../state";
+import FileDetail from "../../../entity/FileDetail";
+import styles from "./MoveFileManifest.module.css";
+import FileSelection from "../../../entity/FileSelection";
+
+/**
+ * Modal overlay for displaying details of selected files for NAS cache operations.
+ */
+export default function MoveFileManifest({ onDismiss }: ModalProps) {
+    // const dispatch = useDispatch(); //TODO: add onMove functionality
+    const fileSelection = useSelector(
+        selection.selectors.getFileSelection,
+        FileSelection.selectionsAreEqual
+    );
+    const moveFileTarget = useSelector(interaction.selectors.getMoveFileTarget);
+
+    const [fileDetails, setFileDetails] = React.useState<FileDetail[]>([]);
+    const [totalSize, setTotalSize] = React.useState<number>(0);
+
+    React.useEffect(() => {
+        async function fetchDetails() {
+            const details = await fileSelection.fetchAllDetails();
+            setFileDetails(details);
+            const totalFileSize = details.reduce((acc, file) => acc + (file.size || 0), 0);
+            setTotalSize(totalFileSize);
+        }
+
+        fetchDetails();
+    }, [fileSelection]);
+
+    const onMove = () => {
+        console.log(
+            `Moving ${fileDetails.length} files ${
+                moveFileTarget === "ON_TO_NAS" ? "onto" : "off of"
+            } NAS.`
+        );
+        onDismiss();
+    };
+
+    const body = (
+        <div>
+            <p>Selected Files:</p>
+            <div className={styles.fileListContainer}>
+                <ul className={styles.fileList}>
+                    {fileDetails.map((file) => (
+                        <li key={file.id} className={styles.fileItem}>
+                            {file.name} - {formatFileSize(file.size || 0)}
+                        </li>
+                    ))}
+                </ul>
+            </div>
+            <p>Total Files: {fileDetails.length}</p>
+            <p>Total Size: {formatFileSize(totalSize)}</p>
+        </div>
+    );
+
+    return (
+        <BaseModal
+            body={body}
+            footer={
+                <PrimaryButton
+                    className={styles.confirmButton}
+                    disabled={!fileDetails.length}
+                    iconName="Accept"
+                    onClick={onMove}
+                    text="CONFIRM"
+                    title="Confirm"
+                />
+            }
+            onDismiss={onDismiss}
+            title={`Move Files ${moveFileTarget === "ON_TO_NAS" ? "onto" : "off of"} NAS Cache`}
+        />
+    );
+}
+
+/**
+ * Formats a file size to a human-readable string.
+ */
+const formatFileSize = (size: number) => {
```

Comment:
This should go after the imports, but I think we have a function for doing this sort of thing already. Look in the aggregate file box, that display already renders the appropriate byte unit.

Reply:
resolved in [ae06c75](https://github.com/AllenInstitute/biofile-finder/pull/311/commits/ae06c75ecf44f37d7f577e30a61ce146c19f0d27)